app/src/main/kotlin/com/wire/android/util/FileUtil.kt

```diff
@@ -204,12 +202,15 @@ fun Context.startMultipleFileSharingIntent(path: String) {
 fun openAssetFileWithExternalApp(assetDataPath: Path, context: Context, assetExtension: String, onError: () -> Unit) {
     try {
         val assetUri = context.pathToUri(assetDataPath)
-
+        val mimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(assetExtension)
         // Set intent and launch
         val intent = Intent()
-        intent.setActionViewIntentFlags()
-        intent.setDataAndType(assetUri, MimeTypeMap.getSingleton().getMimeTypeFromExtension(assetExtension))
-
+        intent.apply {
+            action = Intent.ACTION_VIEW
+            addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
```

Comment:
Shouldn't we also add `Intent.FLAG_GRANT_WRITE_URI_PERMISSION` for versions below `LOLLIPOP`?

Reply:
The min API level is 24 and LOLLIPOP is 21, so we don't really need it

Reply:
True that. I forgot our min was 24 😅